### PR TITLE
Adding endpoint_credentials param and changed Slice to SliceStable for sorting. 

### PR DESCRIPTION
**Pull request by kajeeth**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -165,6 +167,14 @@ func newCloudWatchLogsClient(roleARN string, sess *session.Session, endpoint str
 		}
 		svcConfig.EndpointResolver = endpoints.ResolverFunc(cwCustomResolverFn)
 	}
+	if credsEndpoint != "" {
+		logrus.Infof("Trying to get credentials from credentials_endpoint %s", credsEndpoint)
```
Comment (PettitWesley) [inline cloudwatch/cloudwatch.go]: This info statement is redundant since you already print it when you read the parameter.

Comment (kajeeth) [reply cloudwatch/cloudwatch.go]: Yeah, makes sense. will remove it

File: fluent-bit-cloudwatch.go [hunk 2]
```diff
@@ -83,6 +83,8 @@ func getConfiguration(ctx unsafe.Pointer, pluginID int) cloudwatch.OutputPluginC
 	logrus.Infof("[cloudwatch %d] plugin parameter auto_create_group = '%v'\n", pluginID, config.AutoCreateGroup)
 	config.CWEndpoint = output.FLBPluginConfigKey(ctx, "endpoint")
 	logrus.Infof("[cloudwatch %d] plugin parameter endpoint = '%s'\n", pluginID, config.CWEndpoint)
+	config.CredsEndpoint = output.FLBPluginConfigKey(ctx, "credentials_endpoint")
+	logrus.Info("[cloudwatch %d] plugin parameter credentials_endpoint = %s\n", pluginID, config.CredsEndpoint)
```
Comment (PettitWesley) [inline fluent-bit-cloudwatch.go]: This should be `Infof`

**Comment by PettitWesley** [comment]
Ran the cloudwatch integ tests; they passed.

File: fluent-bit-cloudwatch.go (hunk 2 above)
Comment (kajeeth) [reply fluent-bit-cloudwatch.go]: Thanks!
Will make both the changes and create a new revision

**Pull request merged**